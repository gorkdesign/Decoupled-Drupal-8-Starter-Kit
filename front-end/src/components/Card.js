import React from 'react';
import PropTypes from 'prop-types';
import MdRemove from 'react-icons/lib/md/remove-circle-outline';

const Card = (props) => {
  const { node, ctaHandler, deleteHandler } = props;
  return (
    <div className="py-3">
      <div
        role="button"
        tabIndex={0}
        className="card"
        onClick={() => { ctaHandler(node); }}
        onKeyUp={() => { ctaHandler(node); }}
      >
        <div className="row ">
          <div className="col-md-4">
            { node.images.length === 0 ?
              <img src="https://placeholdit.imgix.net/~text?txtsize=38&amp;txt=¯\_(ツ)_/¯&amp;w=400&amp;h=400" alt="" className="w-100" />
                :
              <img src={node.images[0].url} alt="" className="w-100" />
              }
          </div>
          <div className="col-md-8 px-3">
            <div className="card-body px-3">
              <h4 className="card-title">{node.title === 'NULL' ? 'NO TITLE' : node.title }</h4>
              {node.body !== null ?
                <p className="card-text">{node.body.value}</p> : null }
              <p className="card-text">NID: {node.nid}</p>
              <button
                className="delete"
                onClick={deleteHandler}
                data-toggle="modal"
                data-target="#exampleModalCenter"
              >
                <MdRemove className="remove" />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

Card.propTypes = {
  // node: PropTypes.shape({
  //   author: PropTypes.shape({
  //     name: PropTypes.string,
  //   }),
  //   body: {
  //     value: PropTypes.string,
  //   },
  //   images: PropTypes.arrayOf(PropTypes.shape({
  //     url: PropTypes.string,
  //     mid: PropTypes.number,
  //   })),
  //   nid: PropTypes.number,
  //   title: PropTypes.string,
  //   uuid: PropTypes.string,
  // }).isRequired,
  ctaHandler: PropTypes.func.isRequired,
  deleteHandler: PropTypes.func.isRequired,
};

export default Card;
