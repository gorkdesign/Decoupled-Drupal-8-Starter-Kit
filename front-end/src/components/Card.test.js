import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Card from './Card';

const node = {
  images: [],
  title: 'test',
  body: {
    value: 'test'
  }
};

const generateShallowCard = (node) => {
  return shallow(
    <Card node={node} deleteHandler={jest.fn()} ctaHandler={jest.fn()} />
  );
}


describe('Card', () => {

  it('should render correctly', () => {
    const output = generateShallowCard(node);
    expect(shallowToJson(output)).toMatchSnapshot();
  });

});