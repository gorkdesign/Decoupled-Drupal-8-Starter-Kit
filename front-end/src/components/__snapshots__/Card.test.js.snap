// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<div
  className="py-3"
>
  <div
    className="card"
    onClick={[Function]}
  >
    <div
      className="row "
    >
      <div
        className="col-md-4"
      >
        <img
          alt=""
          className="w-100"
          src="https://placeholdit.imgix.net/~text?txtsize=38&txt=¯\\\\_(ツ)_/¯&w=400&h=400"
        />
      </div>
      <div
        className="col-md-8 px-3"
      >
        <div
          className="card-body px-3"
        >
          <h4
            className="card-title"
          >
            test
          </h4>
          <p
            className="card-text"
          >
            test
          </p>
          <p
            className="card-text"
          >
            NID: 
          </p>
          <button
            className="delete"
            data-target="#exampleModalCenter"
            data-toggle="modal"
            onClick={[Function]}
          >
            <MdRemoveCircleOutline
              className="remove"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
